src/distrs/signrank.jl

```diff
@@ -0,0 +1,109 @@
+#=
+The signrank distribution is equivalent to the problem
+of finding the number of subsets of {1,2,...,n} summing to W,
+relative to the total number of subsets.
+The empty subset is defined to sum to zero.
+This can be calculated using the recursion:
+either n is in the subset in which case we need to calculate
+the number of subsets of {1,2,...,n-1} summing to W-n,
+or n is not in the subset in which case we need to calculate
+the number of subsets of {1,2,...,n-1} summing to W.
+This can be calculated bottom up using dynamic programming.
+
+The i'th element of DP in the j'th outer loop iteration represents:
+the number of ways {1,2,...,j} can sum to W-i+1.
+ =#
+
+@inline function signrankDP(n, W)
+    DP = zeros(Int, W + 1)
+    DP[W+1] = 1
+    for j in 1:n
+        for i in 1:(W+1-j)
+            DP[i] += DP[i+j]
+        end
+    end
+    return DP
+end
+
+function signrankpdf(n::Int, W::Union{Float64,Int})
+    f, _ = modf(W)
+    if f != 0
+        return 0.0
+    end
+    W = Int(W)
+    max_W = (n * (n + 1)) >> 1
+    if W < 0
+        return 0.0
+    elseif W > max_W >> 1
+        return signrankpdf(n, max_W - W)
+    end
```

Comment:
IMO a bit clearer since it's a bit more obvious that computations are always carried out for the minimum of W and max_W - W. Also slightly optimized since max_W does not have to be computed for W < 0:
```suggestion
    if W < 0
        return 0.0
    end
    max_W = (n * (n + 1)) >> 1
    W2 = max_W - W
    if W2 < W
        return signrankpdf(n, W2)
    end
```